nestkernel/conn_builder.cpp

```diff
@@ -99,17 +100,29 @@ nest::ConnBuilder::ConnBuilder( Network& net,
   if ( !default_weight_and_delay_ )
   {
     weight_ = syn_spec->known( names::weight )
-      ? ConnParameter::create( ( *syn_spec )[ names::weight ] )
-      : ConnParameter::create( ( *syn_defaults )[ names::weight ] );
+      ? ConnParameter::create( ( *syn_spec )[ names::weight ], net_.get_num_threads() )
+      : ConnParameter::create( ( *syn_defaults )[ names::weight ], net_.get_num_threads() );
+    // If weight is an array it will be added to a vector of ConnParameters,
+    // ensuring thread-safety of the connection routine.
```

Comment:
This comment does not do much more than restate what the code already tells, particularly the first line. It is also, just as the code, repeated several times. In true Robert C Martin "Clean Code" spirit, we could delegate the task here:
```
register_parameters_requiring_skipping_( *weight_ );
```
with
```
void register_parameters_requiring_skipping_( ConnParam& param )
{
  if ( param.is_array() )
  {
    skip_array_parameters_.push_back( &param );
  }
}
```

And (again in total RC Martin mode), one could change `skip_array_parameters` to `parameters_requiring_skipping_`.